km62/Kondratiuk_Andrii/laboratory1.sql

```diff
@@ -1,2 +1,119 @@
 -- LABORATORY WORK 1
 -- BY Kondratiuk_Andrii
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+модифікувати таблиці та вставляти дані
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+--SQL USER
+CREATE USER kondratyuk IDENTIFIED BY 1111
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP"
+--QUOUTE
+ALTER USER kondratyuk QUOTE 100M ON "USERS"
+--CONNECTION
+GRANT CONNECT TO studentpma
+--GRANTED
+INSERT ANY TABLE
+ALTER ANY EDITION;
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент купляє квиток на потяг.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE TABLE STUDENT
+    (student_name VARCHAR2(30) NOT NULL);
+ALTER TABLE STUDENT
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+GRANT CREATE ANY TABLE
+GRANT SELECT ANY TABLE
+GRANT ALTER ANY TABLE
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Якого товару найменше продано?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+SELECT PROD_ID, MIN(ORDER_ITEM) AS res_quant FROM ORDERITEMS
+GROUP BY PROD_ID;
+
+
```

Comment:
PROJECT ORDERITEMS TIMES PRODUCTS
WHERE (ORDERITEMS.QUANTITY=(PROJECT(ORDERITEMS){MIN(QUANTITY)})
AND 
ORDERITEMS.PROD_ID = PRODUCTS.PROD_ID))
{PRODUCTS.PROD_NAME)